### ISSUE TITLE
Can't call dall-e-3

### ISSUE THREAD

**Issue by Tameflame** [tjardoo/openai-client#39]
I get the following error:
thread 'tokio-runtime-worker' panicked at src/ai_api/api.rs:56:59:
called `Result::unwrap()` on an `Err` value: ParseError("invalid value: map, expected map with a single key")

The relevant function:

```
pub async fn create_image_with_prompt(prompt: &str) -> String {

    let env_variables = read_file(".env").unwrap();
    let api_key: String = (&env_variables["OPENAI_API_KEY"]).to_owned();

    let client = Client::new(api_key);
    let parameters = CreateImageParameters {
        prompt: "A cute baby dog".to_string(),
        model: Some("dall-e-3".to_string()),
        n: Some(1),
        quality: None,
        response_format: Some(ResponseFormat::Url),
        size: Some(ImageSize::Size1024X1024),
        style: None,
        user: None,
    };

    let result = client.images().create(parameters).await.unwrap();
    let paths = result.save("./images").await.unwrap();
    let models_result = client.models().list().await.unwrap();

    println!("{:#?}", models_result);
    println!("{:?}", paths);

    println!("{:#?}", result);
    return "request complete".to_string();
}
```

**Comment by tjardoo** [comment]
Thanks for reporting! This issue should now be fixed with #41

**Comment by Tameflame** [comment]
Thanks! Ill let you know if there are any issues

**Comment by Tameflame** [comment]
@tjardoo 

Sorry but I still get the same error after upgrading to the latest version of this package (0.3.6):

My code:

```
pub async fn create_image_with_prompt(prompt: &str) -> String {

    let env_variables = read_file(".env").unwrap();
    let api_key: String = (&env_variables["OPENAI_API_KEY"]).to_owned();

    let client = Client::new(api_key);
    let parameters = CreateImageParameters {
        // prompt: prompt.to_string(),
        prompt: "Create a dragon monster in the style of digimon".to_string(),
        model: Some("dall-e-3".to_string()),
        n: Some(1),
        quality: Some("hd".to_string()),
        response_format: Some(ResponseFormat::Url),
        size: Some(ImageSize::Size1024X1024),
        style: Some(ImageStyle::Vivid),
        user: None,
    };

    let result = client.images().create(parameters).await.unwrap();
    let paths = result.save("./images").await.unwrap();

    return "hello world".to_string();
}
```

The error:

`called `Result::unwrap()` on an `Err` value: ParseError("invalid value: map, expected map with a single key")`

Simply changing dall-e-3 to dall-e-2 works, so it's something specific to calling dall-e-3

**Comment by tjardoo** [comment]
@Tameflame this issue should now be fixed with #43. `dall-e-2` and `dall-e-3` both had a different response format - now both are supported. Let's me know if there are any other issues. Thanks!
